utils/dockerutil.py

```diff
@@ -177,6 +177,27 @@ def find_cgroup(self, hierarchy):
             raise Exception("Can't find mounted %s cgroups." % hierarchy)
 
     @classmethod
+    def find_cgroup_from_proc(cls, mountpoints, pid, subsys):
+        with open('/proc/{pid}/cgroup'.format(pid=pid), 'r') as fp:
```

Comment:
watchout you'll need to use `self._docker_root` here otherwise this won't work if the agent runs in a container where `/proc` is mounted on `/host/proc`